### PR TITLE
docs($api/module-methods): add eager-weak + notes to resolveWeak

### PR DESCRIPTION
**Pull request by faceyspacey**
Docs update corresponding to:

https://github.com/webpack/webpack/pull/5235

### CODE REVIEW

**Comment by skipjack** [comment]
@faceyspacey thanks for remembering to add documentation for this new feature. Let's wait until webpack/webpack#5235 is merged and released to merge and release this. Once the main PR is merged we should probably ping a few others for review.

File: content/api/module-methods.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ import(
...
 - `"eager"`: Generates no extra chunk. All modules are included in the current chunk and no additional network requests are made. A `Promise` is still returned but is already resolved. In contrast to a static import, the module isn't executed until the call to `import()` is made.
+- `"eager-weak"`: Generates chunks identically to `"lazy"`, but leaves the responsibility of serving the chunks to the client up to the user. A `Promise` is still returned but, only successfully resolves if the chunks are already on the client. A network request will never be performed. This is useful for universal rendering when required chunks are always manually served in initial requests (embedded within the page), but not in cases where app navigation will trigger an import not initially served.
```
Comment (sokra) [inline content/api/module-methods.md]: `"weak"`: Tries to load the module if the module function has already been loaded in some other way (i. e. another chunk imported it or a script containing the module was loaded). A `Promise` is still returned but, only successfully resolves if the chunks are already on the client. If the module is not available, the Promise is rejected. A network request will never be performed. This is useful for universal rendering when required chunks are always manually served in initial requests (embedded within the page), but not in cases where app navigation will trigger an import not initially served.

**Comment by faceyspacey** [comment]
@sokra's requested change has been made

**Comment by faceyspacey** [comment]
@sokra's requested changes have been applied

**Comment by skipjack** [comment]
@faceyspacey was this released yet? No rush, but ping me when it is so we can get this in.

**Comment by skipjack** [comment]
Ok it seems the corresponding PR was merged a couple weeks ago and a few releases have been done since then. Merging...

**Pull request merged**